src/compiler/generate-common-tests.xsl

```diff
@@ -721,19 +735,40 @@
          else QName($thisvar/@namespace-uri, $thisvar/@name)
          "/>
       <xsl:variable name="distinctqnames" as="xs:QName*" select="distinct-values($qnames)"/>
-      <!-- Return distinct strings, except that any unprefixed name with nonempty URI
-         needs Q{} notation to record the namespace URI. -->
-      <xsl:sequence select="for $thisqname in $distinctqnames return
-         if ( empty(prefix-from-QName($thisqname)) and (string-length(namespace-uri-from-QName($thisqname)) gt 0) )
-         then concat('Q{',namespace-uri-from-QName($thisqname),'}',local-name-from-QName($thisqname))
-         else string($thisqname)
-         "/>
+      <!-- Return distinctly named <x:var> elements. Any unprefixed name with nonempty URI
+         or any prefixed name that is not uniquely bound in the set of variables
+         uses Q{} notation to record the namespace URI. -->
+      <xsl:for-each select="$distinctqnames">
+         <xsl:variable name="thisqname" select="."/>
+         <xsl:variable name="thisprefix" select="prefix-from-QName($thisqname)"/>
+         <xsl:choose>
+            <xsl:when test="empty(prefix-from-QName($thisqname)) and (string-length(namespace-uri-from-QName($thisqname)) gt 0)">
+               <!-- No prefix but there is a nonempty namespace URI -->
+               <x:var>
+                  <xsl:attribute name="name" select="concat('Q{',namespace-uri-from-QName($thisqname),'}',local-name-from-QName($thisqname))"/>
+               </x:var>
+            </xsl:when>
+            <xsl:when test="string-length(namespace-uri-from-QName($thisqname)) eq 0">
+               <!-- No namespace -->
+               <x:var>
+                  <xsl:attribute name="name" select="local-name-from-QName($thisqname)"/>
+               </x:var>
+            </xsl:when>
+            <xsl:otherwise>
+               <!-- Prefix bound to a namespace -->
+               <x:var>
+                  <xsl:namespace name="{$thisprefix}" select="namespace-uri-from-QName($thisqname)"/>
+                  <xsl:attribute name="name" select="$thisqname"/>
+               </x:var>
+            </xsl:otherwise>
+         </xsl:choose>
```

Comment:
In this `xsl:choose`, every conditional branch is `<x:var></x:var>`. Then it could be consolidated. i.e. `<x:var><xsl:choose></xsl:choose></x:var>`.

Reply:
OK, I did that in 21ed0e1f844324b13670d7ad7828bb9894d2a26f.